### PR TITLE
Update Coding Guidlines

### PR DESCRIPTION
**Pull request by iSazonov**
## PR Summary

Update Coding guidelines.

## PR Context  

We always prefer overloads with explicit parameters for formatting and comparing so add this in docs.

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Change is not breaking](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.  
- **User-facing changes**
    - [x] Not Applicable
    - **OR**  
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [x] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
        - [ ] [Add `[feature]` to your commit messages if the change is significant or affects feature tests](https://github.com/PowerShell/PowerShell/blob/master/docs/testing-guidelines/testing-guidelines.md#requesting-additional-tests-for-a-pr)

### CODE REVIEW

File: docs/dev-process/coding-guidelines.md [hunk 1]
```diff
@@ -108,6 +108,9 @@ Some general guidelines:
 * Be aware of APIs such as `String.Split(params char[])` that do not provide overloads to avoid array allocation.
   When calling such APIs, reuse a static array when possible (e.g. `Utils.Separators.Colon`).
 
+* Avoid using string interpolations and overloads with implicit parameters like `Culture` and `StringComparison`.
+  Instead, use overloads with explicit parameters like `String.Format(IFormatProvider, String, Object[])` and `Equals(String, String, StringComparison)`.
```
Comment (RDIL) [inline docs/dev-process/coding-guidelines.md]:
```suggestion
  Instead, use overloads with explicit parameters such as `String.Format(IFormatProvider, String, Object[])` and `Equals(String, String, StringComparison)`.
```
`like` does not sound very professional

Comment (iSazonov) [reply docs/dev-process/coding-guidelines.md]: I am not native English. I used the pattern from the file. Could you please review the file and fix issues?

Comment (RDIL) [reply docs/dev-process/coding-guidelines.md]: Sure

File: docs/dev-process/coding-guidelines.md [hunk 2]
```diff
@@ -108,6 +108,9 @@ Some general guidelines:
 * Be aware of APIs such as `String.Split(params char[])` that do not provide overloads to avoid array allocation.
   When calling such APIs, reuse a static array when possible (e.g. `Utils.Separators.Colon`).
 
+* Avoid using string interpolations and overloads with implicit parameters like `Culture` and `StringComparison`.
```
Comment (RDIL) [inline docs/dev-process/coding-guidelines.md]:
```suggestion
* Avoid using string interpolations and overloads with implicit parameters such as `Culture` and `StringComparison`.
```

File: docs/dev-process/coding-guidelines.md [hunk 3]
```diff
@@ -108,6 +108,9 @@ Some general guidelines:
...
+  Instead, use overloads with explicit parameters such as `String.Format(IFormatProvider, String, Object[])` and `Equals(String, String, StringComparison)`.
```
Comment (RDIL) [inline docs/dev-process/coding-guidelines.md]:
```suggestion
  Instead, use overloads with more explicit parameters such as `String.Format(IFormatProvider, String, Object[])` and `Equals(String, String, StringComparison)`.
```

**Review by RDIL — COMMENTED** [review]
This is what I found.

**Review by SteveL-MSFT — APPROVED** [review]

**Review by adityapatwardhan — APPROVED** [review]

**Review by TravisEz13 — APPROVED** [review]

**Pull request merged**